sirepo/package_data/static/js/flash.js

```diff
@@ -9,8 +9,24 @@ SIREPO.app.config(function() {
     SIREPO.SINGLE_FRAME_ANIMATION = ['gridEvolutionAnimation'];
 });
 
-SIREPO.app.factory('flashService', function(appState) {
+SIREPO.app.factory('flashService', function(appState, $rootScope) {
     var self = {};
+    self.originalGridgeometryEnum = [];
+    self.capLaserBELLAGridgeometyEnum = [];
+
+    function initGridgeometry() {
+        if (self.originalGridgeometryEnum.length) {
+            return;
+        }
+        var e = SIREPO.APP_SCHEMA.enum.Gridgeometry;
+        self.originalGridgeometryEnum = e;
+        e.forEach(function(v) {
```

Comment:
Rather than alter the schema, it is cleaner to use PanelState.showEnum()

Reply:
I'll make this change and merge with the pull request along with a few other changes